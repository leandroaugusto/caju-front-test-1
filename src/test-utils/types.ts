export interface ITestUtils {
  children?: React.ReactNode
}
