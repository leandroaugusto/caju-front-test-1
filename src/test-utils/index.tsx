import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";

import { ITestUtils } from "./types";

import theme from "~/theme";

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
