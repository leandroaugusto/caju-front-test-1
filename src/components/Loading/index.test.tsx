import { customRender, screen } from "~/test-utils";

import { Loading } from ".";

describe("Header", () => {
  it("Should render loading component", () => {
    const { container } = customRender(<Loading />);
    const loading = screen.getByTestId("loading-container");

    expect(loading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
