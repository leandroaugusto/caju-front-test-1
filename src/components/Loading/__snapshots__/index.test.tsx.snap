// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Should render loading component 1`] = `
<div>
  <div
    aria-live="polite"
    class="sc-braxZu kSlxkK"
    data-testid="loading-container"
    role="status"
  >
    <div
      class="sc-gJhJTp cNRuoW"
    />
  </div>
</div>
`;
